libsolidity/ast/AST.h

```diff
@@ -510,6 +510,9 @@ class ContractDefinition: public Declaration, public StructurallyDocumented
 
 	std::string fullyQualifiedName() const { return sourceUnitName() + ":" + name(); }
 
+	std::string yulCreationObjectName() const;
+	std::string yulRuntimeObjectName() const;
```

Comment:
These functions seem to be Yul specific, but this is the Solidity AST. Would it make sense to have them as Yul utils instead?

Reply:
I agree, can you suggest some place where I can move them. I need them in both `IRGenerator` and `IRGeneratorForStatements` so will move to `IRGenerationContext` for now, as I see some similar utilities over there